### DISCUSSION TITLE
Monitoring On/Off Preference

### DISCUSSION THREAD

**Discussion by vmitchell85** [KevinBatdorf/laravel-forge-raycast#3] [Ideas]
Personally, I don't really think I'd use the monitoring feature. With other providers for monitoring like Oh Dear I wouldn't use it.

Maybe adding an on/off option in the preferences would be useful in my case?

**Comment by KevinBatdorf** [comment]
You mean to leave on the API request to Forge, but disable the HEAD request? Or disable them both and just have it fetch once on first render?

**Reply by vmitchell85** [reply]
Let me check out the code a bit more to make sure I understand fully what you're doing. But you said you'd be hitting the API every 2-3 seconds so I assumed it was for monitoring the status of the servers.

I mostly use Alfred for quickly opening Forge pages. If I can fully switch to RayCast (unable to yet) maybe that would change.

**Reply by KevinBatdorf** [reply]
Yeah it will ping the Forge API every 2-3 seconds. The idea is to get the deploy status, which I personally use a lot. I'd like to know when it's done deploying without having to open forge.laravel.com. So with this I can just leave the UI open, or check back every now and then.

But it also sends a simple HEAD request to the server (or an alias) to just check it's returning a 200. That's less useful, but it's useful for me when i want to restart a server. I do that sometimes while developing. 

I could maybe make the 2-3 seconds part a setting. Something like "max requests per minute"

**Reply by vmitchell85** [reply]
Based on the code it looks like you're pinging the API every 2-3 seconds once you're viewing a specific server correct? If that is all it's doing then that seems fine.

Would it be lacking something if you only did the checks after a deployment or a reboot? Then you don't have to check all the time but can just check after commands that would affect the status are run.

**Reply by KevinBatdorf** [reply]
> Based on the code it looks like you're pinging the API every 2-3 seconds once you're viewing a specific server correct?

That's correct. The initial server list is requested once when you start using the extension. Then any polling happens when you drill down/ click through to view sites, etc.

> Would it be lacking something if you only did the checks after a deployment or a reboot? 

Possibly, because something else could trigger a deployment. Like another team member or a GitHub action, etc. Probably an edge case though that someone else triggers it just as you open the tool though :) 

It could be a good optimization, but I'm also not sure it's worth implementing to save a few network requests either. Something to consider later perhaps. It's still kind of in MVP stage now I guess you could say.
